src/openaq/incremental_downloader_parallel.py

```
@@ -516,7 +525,8 @@ def download_country_all(self, country_code: str, country_id: int,
                     raise
                 
                 i = batch_end
-                self.save_checkpoint(country_code, i, total_locations, completed_locations, str(output_path))
+                # Save actual progress (start_index + current position)
+                self.save_checkpoint(country_code, start_index + i, total_locations, completed_locations, str(output_path))
```

Pylint (3.3.7)

[convention] Line too long (123/100)

(C0301)